### PR TITLE
slurm: add lz4 dependency

### PR DESCRIPTION
**Pull request by bhipple**
Slurm looks for lz4 by default, and if it finds it can improve its compression techniques

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by bhipple** [comment]
@GrahamcOfBorg build slurm

**Comment by bhipple** [comment]
CC @markuskowa @jagajaga

**Comment by markuskowa** [comment]
What feature does adding `lz4` activate?

File: pkgs/servers/computing/slurm/default.nix [hunk 1]
```diff
@@ -35,15 +35,16 @@ stdenv.mkDerivation rec {
...
       lua hwloc numactl readline freeipmi
   ] ++ stdenv.lib.optionals enableX11 [ libssh2 xorg.xauth ];
 
   configureFlags = with stdenv.lib;
-    [ "--with-munge=${munge}"
-      "--with-ssl=${openssl.dev}"
+    [ "--with-freeipmi=${freeipmi}"
       "--with-hwloc=${hwloc.dev}"
-      "--with-freeipmi=${freeipmi}"
+      "--with-lz4=${lz4}"
```
Comment (markuskowa) [inline pkgs/servers/computing/slurm/default.nix]: Using the dev output here works: `--with-lz4=${lz4.dev}`. 
Would you mind also adding `"--with-zlib=${zlib}"` ?

**Comment by bhipple** [comment]
@GrahamcOfBorg test slurm

**Comment by bhipple** [comment]
> What feature does adding lz4 activate (other than compression for sbcast)?

That's the only one I can see, but I'll admit I have an ulterior motive: I'm trying to compile this pkg on RedHat without sandboxes, and without specifying the NixPkg `lz4` it's finding the system one and getting itself confused half-way through the build.  Strangely it succeeded with a mix of system and nix lz4's when I added the configure flag and didn't have `lz4.dev`, but I've now also built it successfully at home on NixOS.

**Comment by markuskowa** [comment]
From my side this PR is good to be merged.

**Comment by bhipple** [comment]
Ping @Mic92 mind taking another look?  I think this is ready to be merged.

**Pull request merged**